analysis/src/CompletionFrontEnd.ml

```diff
@@ -666,6 +667,24 @@ let completionWithParser ~debug ~path ~posCursor ~currentFile ~text =
       | _ -> ());
     Ast_iterator.default_iterator.typ iterator core_type
   in
+  let pat (iterator : Ast_iterator.iterator) (pat : Parsetree.pattern) =
+    if pat.ppat_loc |> Loc.hasPos ~pos:posNoWhite then (
+      found := true;
+      if debug then
+        Printf.printf "posCursor:[%s] posNoWhite:[%s] Found pattern:%s\n"
+          (Pos.toString posCursor) (Pos.toString posNoWhite)
+          (Loc.toString pat.ppat_loc);
+      (match pat.ppat_desc with
+      | Ppat_construct (lid, _) ->
+        let lidPath = flattenLidCheckDot lid in
+        if debug then
+          Printf.printf "XXX Ppat_construct %s:%s\n"
+            (lidPath |> String.concat ".")
+            (Loc.toString lid.loc);
+        setResult (Cpath (CPId (lidPath, Value)))
```

Comment:
Triggering completion as a `Value`. There's no need for specific completions for `Constructor` because of the upper case. Lower-case ids are normal values, and upper-case values are constructors.